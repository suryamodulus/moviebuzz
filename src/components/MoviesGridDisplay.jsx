import MoviesGridItem from "./MoviesGridItem";

const MoviesGridDisplay = ({ pages }) => {
  return (
    <div className="grid grid-cols-2 md:grid-cols-4 xl:grid-cols-6 gap-0">
      {pages.map((page) => {
        return page.results.map((movie) => {
          return <MoviesGridItem key={movie.id} movie={movie}></MoviesGridItem>;
        });
      })}
    </div>
  );
};

export default MoviesGridDisplay;
