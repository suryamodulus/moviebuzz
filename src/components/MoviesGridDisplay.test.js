import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import MoviesGridDisplay from "./MoviesGridDisplay";

const pagesMockData = [
  {
    results: [
      {
        id: 675353,
        original_title: "Sonic the Hedgehog 2",
        overview:
          "After settling in Green Hills, Sonic is eager to prove he has what it takes to be a true hero. His test comes when Dr. Robotnik returns, this time with a new partner, Knuckles, in search for an emerald that has the power to destroy civilizations. Sonic teams up with his own sidekick, Tails, and together they embark on a globe-trotting journey to find the emerald before it falls into the wrong hands.",
        popularity: 9124.409,
        poster_path: "/egoyMDLqCxzjnSrWOz50uLlJWmD.jpg",
        release_date: "2022-03-30",
        runtime: 122,
        title: "Sonic the Hedgehog 2",
        vote_average: 7.7,
      },
    ],
  },
];

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <MoviesGridDisplay pages={pagesMockData} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
