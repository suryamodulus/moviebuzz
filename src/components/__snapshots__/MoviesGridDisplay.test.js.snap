// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="grid grid-cols-2 md:grid-cols-4 xl:grid-cols-6 gap-0"
>
  <a
    className="flex"
    href="/movie/675353"
    onClick={[Function]}
  >
    <img
      alt="Sonic the Hedgehog 2"
      className="w-full"
      src="https://image.tmdb.org/t/p/w185//egoyMDLqCxzjnSrWOz50uLlJWmD.jpg"
    />
  </a>
</div>
`;
